scipy/stats/tests/test_distributions.py

```diff
@@ -863,6 +863,20 @@ def test_logpdf(self):
         assert_(np.isfinite(b.logpdf(x)).all())
         assert_allclose(b.pdf(x), np.exp(b.logpdf(x)))
 
+    def test_cdf(self):
+        # regression test for gh-4030: Implementation of 
+        # scipy.stats.betaprime.cdf()
+        x = stats.betaprime.cdf(0, 0.2, 0.3)
+        assert_equal(x, 0.0)
+
+        alpha, beta = 267, 1472
+        x = np.array([0.2, 0.5, 0.6])
+        cdfs = stats.betaprime.cdf(x, alpha, beta)
+        assert_(np.isfinite(cdfs).all())
+        
+        # check the new cdf implementation vs generic one:
+        cdfs_g = stats.rv_continuous.cdf(stats.betaprime, x, alpha, beta)
```

Comment:
Aaargh, you're right of course. 
Bent over backwards, in the force-pushed revision.